src/testrender/optixraytracer.cpp

```diff
@@ -359,7 +1001,125 @@ OptixRaytracer::finalize_scene()
     m_optix_ctx["output_buffer"]->set(buffer);
 
     m_optix_ctx->validate();
-#endif
+
+#else //#if (OPTIX_VERSION < 70000)
+
+    // Build acceleration structures
+    OptixAccelBuildOptions accelOptions;
+    OptixBuildInput buildInputs[2];
+
+    memset(&accelOptions, 0, sizeof(OptixAccelBuildOptions));
+    accelOptions.buildFlags = OPTIX_BUILD_FLAG_NONE;
+    accelOptions.operation  = OPTIX_BUILD_OPERATION_BUILD;
+    accelOptions.motionOptions.numKeys = 0;
+    memset(buildInputs, 0, sizeof(OptixBuildInput) * 2);
+
+    // Set up quads input
+    void* d_quadsAabb;
+    std::vector<OptixAabb> quadsAabb;
+    std::vector<QuadParams> quadsParams;
+    quadsAabb.reserve(scene.quads.size());
+    quadsParams.reserve(scene.quads.size());
+    std::vector<int> quadShaders;
+    quadShaders.reserve(scene.quads.size());
+    for (const auto& quad : scene.quads) {
+       OptixAabb aabb;
+       quad.getBounds(aabb.minX, aabb.minY, aabb.minZ,
+                      aabb.maxX, aabb.maxY, aabb.maxZ);
+       quadsAabb.push_back(aabb);
+       QuadParams quad_params;
+       quad.setOptixVariables(&quad_params);
+       quadsParams.push_back(quad_params);
+    }
+    // Copy Quads bounding boxes to cuda device
+    CUDA_CHECK (cudaMalloc (&d_quadsAabb,                   sizeof(OptixAabb) * scene.quads.size()));
+    CUDA_CHECK (cudaMemcpy ( d_quadsAabb, quadsAabb.data(), sizeof(OptixAabb) * scene.quads.size(), cudaMemcpyHostToDevice));
+
+    // Copy Quads to cuda device
+    CUDA_CHECK (cudaMalloc (reinterpret_cast<void **>(&d_quads_list)  ,                     sizeof(QuadParams  ) * scene.quads.size()));
+    CUDA_CHECK (cudaMemcpy (reinterpret_cast<void  *>( d_quads_list)  , quadsParams.data(), sizeof(QuadParams  ) * scene.quads.size(), cudaMemcpyHostToDevice));
+
+    // Fill in Quad shaders
+    CUdeviceptr d_quadsIndexOffsetBuffer;
+    CUDA_CHECK (cudaMalloc (reinterpret_cast<void **>(&d_quadsIndexOffsetBuffer), scene.quads.size() * sizeof(int)));
+
+    unsigned int  quadSbtRecord;
+    quadSbtRecord = OPTIX_GEOMETRY_FLAG_NONE;
```

Comment:
This flag should probably be set to `DISABLE_ANY_HIT` instead of `NONE`. Otherwise visibility depends on the traversal order (which can vary between architectures), and we might end up with "missing" geometry. An alternative is to supply `OPTIX_RAY_FLAG_DISABLE_ANYHIT` to the `optixTraceCall`, but it's generally better for performance to disable any-hit using the geometry flag if possible.